File: src/cleansia_android/core/src/main/java/cz/cleansia/core/consent/SignupConsent.kt
package cz.cleansia.core.consent

/**
 * The GDPR consent kinds the backend records, by their on-the-wire integer
 * (`Cleansia.Core.Domain.Enums.ConsentType`). Each app maps this onto its own
 * OpenAPI-generated enum, whose entries are named `_0`.. `_3`.
 */
enum class SignupConsentType(val wireValue: Int) {
    TermsOfService(0),
    PrivacyPolicy(1),
    MarketingEmails(2),
    DataProcessing(3),
    ;

    companion object {
        fun fromWireValue(wireValue: Int): SignupConsentType? =
            entries.firstOrNull { it.wireValue == wireValue }
    }
}

/**
 * One tick, two records: both signup sentences name the Terms of Service and the
 * Privacy Policy by title. Neither form offers a marketing box, so
 * [SignupConsentType.MarketingEmails] must never appear here — a record nobody
 * ticked is worse than no record at all.
 */
val SIGNUP_TICK_CONSENTS: List<SignupConsentType> =
    listOf(SignupConsentType.TermsOfService, SignupConsentType.PrivacyPolicy)

/** A signup tick waiting for a session it can be recorded against. */
data class PendingSignupConsent(
    val email: String,
    val types: List<SignupConsentType>,
)

File: src/cleansia_ios/CleansiaCore/Sources/CleansiaCore/Consent/SignupConsent.swift
import Foundation

/// The GDPR consent kinds the backend records, by their on-the-wire integer
/// (`Cleansia.Core.Domain.Enums.ConsentType`). The customer app maps this onto its
/// OpenAPI-generated enum, whose cases are named `_0`…`_3`.
public enum SignupConsentType: Int, CaseIterable, Sendable {
    case termsOfService = 0
    case privacyPolicy = 1
    case marketingEmails = 2
    case dataProcessing = 3
}

public extension SignupConsentType {
    /// One tick, two records: both signup sentences name the Terms of Service and the
    /// Privacy Policy by title. Neither form offers a marketing box, so `marketingEmails`
    /// must never appear here — a record nobody ticked is worse than no record at all.
    static let signupTick: [SignupConsentType] = [.termsOfService, .privacyPolicy]
}
